tests/cases/compiler/inferringAnyFunctionType1.ts

```diff
@@ -1,5 +1,7 @@
+// @lib: es2015
+
 function f<T extends { "0": (p1: number) => number }>(p: T): T {

```

Comment:
While I have no idea why anybody would like to write this code... ok 😅 this is definitely valid 😉